strawberryfields/engine.py

```diff
@@ -548,7 +548,9 @@ def device_spec(self):
             self._spec = self._connection.get_device_spec(self.target)
         return self._spec
 
-    def run(self, program: Program, *, compile_options=None, **kwargs) -> Optional[Result]:
+    def run(
+        self, program: Program, *, compile_options=None, recompile=False, **kwargs
```

Comment:
Explicitly adding `recompile=False` here was needed as othwerwise `recompile` was an element in `kwargs`. In `run_async` the `recompile` parameter was separated from `kwargs`, hence `run` needs to be adjusted too in its signature and the way it calls `run_async`.